[nmstate-2.1] bond,bridge,ovs: Add support of `ports` keyword

cathay4t opened the pull request:
In old python API, we use `slaves` keyword which been replaced by `port`
keyword. Some user complained about `plural` vs `singular`, hence we add
support of `ports` here.

Unit test case included.

ffmancera reviewed (changes requested):
I think we should extend this to the python bindings.. also, we could support this in nmstate-1.3 too..

cathay4t committed:
bond,bridge,ovs: Add support of `ports` keyword

In old python API, we use `slaves` keyword which been replaced by `port`
keyword. Some user complained about `plural` vs `singular`, hence we add
support of `ports` for bond, linux bridge, ovs bridge, ovs bond and VRF.

Unit test case included.

Signed-off-by: Gris Ge <fge@redhat.com>
